vcf_zarr_spec.md

```diff
@@ -0,0 +1,122 @@
+# VCF Zarr specification
+
+***Version 0.1***
+
+This document is a technical specification for VCF Zarr, an equivalent of VCF that uses Zarr storage.
+
+This specification depends on definitions and terminology from [The Variant Call Format Specification, VCFv4.3 and BCFv2.2](https://samtools.github.io/hts-specs/VCFv4.3.pdf),
+and [Zarr storage specification version 2](https://zarr.readthedocs.io/en/stable/spec/v2.html).
+
+## Compatibility with VCF and BCF
+
+Any VCF file that can be represented in BCF format can be represented in VCF Zarr.
+Like BCF, VCF Zarr does not support the full set of VCF, since BCF is stricter than VCF around what is permitted in the header regarding field and contig information.
+
+## Overall structure
+
+A VCF Zarr store is composed of a top-level Zarr group that contains VCF header information stored as Zarr group attributes,
+and VCF fields and sample information stored as Zarr arrays.
+
+## VCF Zarr group attributes
+
+The VCF Zarr store contains the following mandatory attributes:
+
+| Key                | Value                                                                                |
+|--------------------|--------------------------------------------------------------------------------------|
+| `vcf_zarr_version` | `0.1`                                                                                |
+| `vcf_header`       | The VCF header from `##fileformat` to `#CHROM` inclusive, stored as a single string. |
+| `contigs`          | A list of strings of the contig IDs in the same order as specified in the header.    |
+| `filters`          | A list of strings of the filters in the same order as specified in the header, except for `PASS`, which is always first. |
+
+The `contigs` attribute plays the same role as the dictionary of contigs in BCF, providing a way of encoding a contig (in the `variant_contig` array) with an integer offset into the `contigs` list.
+
+## VCF Zarr arrays
+
+Each VCF field is stored in a separate Zarr array. This specification only mandates the path, shape, dimension names, and dtype of each array. Other array metadata, including chunks, compression, layout order is not specified here.
```

Comment:
We don't necessarily mandate the dtype, do we? I could perfectly well store genotypes as int16 if I wanted to and it shouldn't break stuff.

Reply:
We don't mandate the size of the dtype (this is explained later in the spec), just the kind (int vs float). I've updated to wording to say "general dtype".

Reply:
That's good - it's an important point to clarify I think, readers won't naturally assume that the width of the basic numeric types (int/float) is flexible.